### PR TITLE
Support appchart customization

### PR DESCRIPTION
**Pull request by andreas-kupries**
Fix: https://github.com/epinio/epinio/issues/1252

- [x] Extend manifest structures with key/value map for customization
- [x] Extend app structures with the same
- [x] Extend appchart API structures with the  user-visible app chart customization settings
- [x] Extend display of appchart details with table of the user-visible app chart customization settings
- [x] Test new display (list, details)
- [x] Add option `--chart-value` (`create`, `update`, `push`). See handling of `--env` for analogous code.
- [x] Implement server-side storage of chart values provided by `--chart-value`. See `--route` for basic analogous code.
- [x] Retrieve chart values when retrieving app data
- [x] Extend display of app details, show chart values (cli side validation for awareness ?)
- [x] Test extended display
- [x] Implement endpoint to validate app customization against app chart declarations
- [x] Test validation
- [x] Add validation of chart values to the app update endpoint as well, to be run when the app is active
- [x] Extend low-level app deployment code to merge chart customization (user-visible, app chart) into the deployed helm chart.
- [x] Test customized app deployment
- [x] Update swagger

### CODE REVIEW

**Comment by andreas-kupries** [comment]
Tech dept reported by linter placed into new ticket #1695

**Comment by andreas-kupries** [comment]
cli/api failures are mostly legit ... cli because the validation is now active, and the test in question used the standard chart, which has now chart values declared for it

**Comment by andreas-kupries** [comment]
:tada: upgrade test fixed ... time to clean the history a bit

**Comment by andreas-kupries** [comment]
Down to the test known to getting stuck.

File: pkg/api/core/v1/client/apps.go
Comment (jimmykarily) [inline pkg/api/core/v1/client/apps.go:279]:
```suggestion
// AppValidateCV uploads validates the chart values of the specified app against its appchart
```

Comment (jimmykarily) [reply pkg/api/core/v1/client/apps.go:279]: uploads or validates? strange syntax there

File: pkg/api/core/v1/models/env.go
Comment (jimmykarily) [inline pkg/api/core/v1/models/env.go:3]: Do we really need to warn people about this?

Comment (jimmykarily) [reply pkg/api/core/v1/models/env.go:3]: There are ways to DRY this if we want to. E.g. we can create a struct that implements all these methods and add it as a "promoted field" both in `EnvVariable` and `AppSetting`

```
type EnvVariable struct {
  KeyValueObject
}

type AppSetting struct {
  KeyValueObject
}
```

though given the only methods/fields we need are those of the new struct, maybe we don't need the wrapping structs at all. 

Let's think about it outside this PR.

File: acceptance/api/v1/appchart_index_test.go
Comment (andreas-kupries) [reply acceptance/api/v1/appchart_index_test.go:38]: Oh, you do not want this comment ? I believe it is sensible to note this, for future maintainers.
I would be ok with just removing the commented `Expect(...)`

File: acceptance/helpers/machine/appcharts.go
Comment (andreas-kupries) [reply acceptance/helpers/machine/appcharts.go:69]: While you can add an `enum` spec to bool parameters validation will ignore it, as it is not sensible

Comment (andreas-kupries) [reply acceptance/helpers/machine/appcharts.go:69]: clarified in the string now

File: internal/api/v1/application/update.go
Comment (andreas-kupries) [reply internal/api/v1/application/update.go:100]: I am torn a bit. `BadRequest` is to me the request just running, i.e. the validation. And that request is not bad per se. The data it is checking is. Maybe make it StatusOK, and a custom response structure carrying the valid/not valid information?

File: acceptance/api/v1/appchart_index_test.go
Comment (jimmykarily) [reply acceptance/api/v1/appchart_index_test.go:38]: So the purpose it to warn people about checking the number of charts which would introduce flakiness. Fine then.

File: internal/api/v1/application/validatecv.go
Comment (andreas-kupries) [reply internal/api/v1/application/validatecv.go:59]: Ok. With the SO references I am becoming OK with changing this to BadRequest.

File: internal/api/v1/application/update.go
Comment (andreas-kupries) [reply internal/api/v1/application/update.go:100]: Ok. With the SO references I am becoming OK with changing this to BadRequest.

Comment (jimmykarily) [reply internal/api/v1/application/update.go:100]: Yes, I'm not 100% sure either. If it was a `push` that also did validation, then it should definitely be a BadRequest. But the validation endpoint validated just fine. I'll leave it to you to decide.

File: internal/helm/helm.go
Comment (andreas-kupries) [reply internal/helm/helm.go:1]: will do

File: internal/helm/helm.go
Comment (andreas-kupries) [reply internal/helm/helm.go:199]: Agreed. To all three of `DRY is complex now`, `Keep it`, and `Retry DRY if this becomes larger`.

File: pkg/api/core/v1/models/env.go
Comment (andreas-kupries) [reply pkg/api/core/v1/models/env.go:3]: > Do we really need to warn people about this?

People in general ? No.
The future maintainers, us or others ? IMHO yes.

Agreed to think in separate issue and PR.

File: pkg/api/core/v1/client/apps.go
Comment (andreas-kupries) [reply pkg/api/core/v1/client/apps.go:279]: Stronger typo. The `uploads` should and will be removed.

File: acceptance/api/v1/application_validate_cv_test.go
Comment (andreas-kupries) [reply acceptance/api/v1/application_validate_cv_test.go:150]: Asked around, yes to `too`. Will fix.

File: internal/api/v1/application/update.go
Comment (andreas-kupries) [reply internal/api/v1/application/update.go:100]: I switched to `BadRequest`.

File: acceptance/api/v1/appchart_index_test.go
Comment (andreas-kupries) [reply acceptance/api/v1/appchart_index_test.go:38]: Correct about the purpose.

Comment (andreas-kupries) [reply acceptance/api/v1/appchart_index_test.go]: Done 3db1362fdea6d67b232b6885ae9926fb37b8fafa

File: acceptance/api/v1/application_validate_cv_test.go
Comment (andreas-kupries) [reply acceptance/api/v1/application_validate_cv_test.go]: Done 4312e15a55163db8d179f3a1a8c40bb8f7a60937

File: internal/api/v1/application/update.go
Comment (andreas-kupries) [reply internal/api/v1/application/update.go:99]: Done 4312e15a55163db8d179f3a1a8c40bb8f7a60937

File: internal/api/v1/application/validatecv.go
Comment (andreas-kupries) [reply internal/api/v1/application/validatecv.go:58]: Done 4312e15a55163db8d179f3a1a8c40bb8f7a60937

File: internal/helm/helm.go
Comment (andreas-kupries) [reply internal/helm/helm.go]: Done b62fee01e46c4ad7fd6620c43a29f57642fcfd98

File: acceptance/helpers/machine/appcharts.go
Comment (andreas-kupries) [reply acceptance/helpers/machine/appcharts.go]: Done fe205197d87f3c8781efbb27a3cc00e4f3caa04c

File: pkg/api/core/v1/client/apps.go
Comment (andreas-kupries) [reply pkg/api/core/v1/client/apps.go]: Done 082bb31a3d9357a6b384c682472c6e3cf85f253c

File: acceptance/apps_test.go
Comment (andreas-kupries) [reply acceptance/apps_test.go]: Done 0d36d98c04eb12f49fa4556addf78aded1c9bcc2

File: internal/helm/helm.go
Comment (andreas-kupries) [reply internal/helm/helm.go:328]: Done 0dc1fa5143b3cd3ee7f9e507b6db3e8c2130f377

**Review by jimmykarily — APPROVED** [review]

**Pull request merged**
